<launch>
  <!-- gps_testing.launch, created by: William Willie Wells, in May 2016 -->

  <!--======================MIT AscTec Protocol and Communication=====================-->
    <!-- movement at constant velocity -->
  <!--include file="$(find collab_launch)/basic_flight/hummingbird_outdoor.launch" /-->
  <include file="$(find gps_testing)/launch/hummingbird_outdoor.launch" />
  <!--include file="$(find gps_testing)/launch/firefly_outdoor.launch" /-->

  <!--============================RF and GPS Communication============================-->
  <include file="$(find gps_testing)/launch/rf_testing.launch" />

  <!--================================Bag Recording===================================-->  
    <!-- custom bag placement, -a -o records a subset, --> 
    <!-- scp -r /path/bagDirectory wwells@cse.unl.edu:/home/research3/elbaum/nimbus/rosBags/gps_testing -->
    <!-- sed -i.bak s/unl_comp_sci#/crash#+1/ path/gps_testing.launch -->

  <!--node pkg="rosbag" name="record" type="record" args="-O $(find gps_testing)/bags/hummingbird/name/hummingbird_name1 /a/cmd_subject_ctrl_state /a/quad_ctrl_input /a/robot_ctrl_input /a/robot_imu /a/robot_poll /a/robot_status /a/subject_gps /a/robot_gps /a/subject_ctrl_state /a/subject_pose /a/subject_status /a/task_waypose /a/launch_waypose /a/pid_input /a/pid_waypose /a/robot_rx_data /a/robot_tx_data"/-->

  <group ns="w">
  <!--node name="flight_logger" pkg="flight_logger" type="flight_logger.py" output="screen" />
  <node name="flight_log_processor" pkg="flight_logger" type="flight_log_processor.py" output="screen" /-->
  <!--include file="$(find collab_launch)/common/record.launch" /-->
  <!-- sed -i.bak s/unl_comp_sci#/crash#+1/ path/bat -->
  <!-- ./path/bat -->

  <!--==============================Arduino Communication============================-->
    <!--node name="arduino_comm" type="serial_comm" pkg="communication">
      <remap from="serial_rx_data" to="device_rx_data"/>
      <remap from="serial_tx_data" to="device_tx_data"/>
      <param name="serial_baudrate" value="9600"/>
      <param name="serial_device" value="/dev/ttyACM0"/> 
    </node>

    <node name="parse_serial" type="parse_serial" pkg="gps_testing" output="screen"></node-->

  <!--====================================Main Node==================================-->
    <!--node name="basic_flight" type="basic_flight" pkg="gps_testing" output="screen"></node>
    <node required="true" name="gps_testing" type="gps_testing" pkg="gps_testing" output="screen">
      <param name="position_change" value="100.0"/> 
      <param name="launch_height" value="0.5"/>
      <param name="lat_or_long" value="true"/>
      <param name="ctrl_rate" value="30"/> 
      <param name="altitude_change" value="0.5"/>
    </node-->

    <!--node name="gps_resume" type="rostopic" pkg="rostopic"
      args="pub -r 30.0 'resume_test' gps_testing/Move
      '{header: auto, enable_gps_test: 1, enable_rf_test: 0, end_test: 0}'"/-->

   <!--======================================GUI====================================-->
   <!--node name="gui_fly" type="gui_fly" pkg="gui_fly">
      <param name="min_y" value="-255.0"/>
      <param name="max_y" value="255.0"/>
      <param name="min_x" value="-255.0"/>
      <param name="max_x" value="255.0"/>
      <param name="min_z" value="-30.0"/>
      <param name="max_z" value="100.0"/>
    </node-->
 
  </group>
</launch>
